src/runtime/auto-prompt-manager.ts

```diff
@@ -230,12 +237,20 @@ export class AutoPromptManager {
       return;
     }
 
-    if (!clientConfig.uiPredicates?.canDisplayAutoPrompt) {
+    const shouldRenderOnsitePreview =
+      article.previewEnabled && this.onsitePreviewEnabled_;
+
+    // If onsite preview should be rendered, canDisplayAutoPrompt will be ignored.
+    if (
+      !shouldRenderOnsitePreview &&
+      !clientConfig.uiPredicates?.canDisplayAutoPrompt
+    ) {
       return;
     }
 
+    // If onsite preview should be rendered, entitlements will be ignored.
     const hasValidEntitlements = entitlements.enablesThis();
```

Comment:
Would it be more readable if we wrapped these checks together so that onsitePreview can skip them all?
```
    if (!shouldRenderOnsitePreview) {
       ...  // Current checks for uiPredicates, entitlements
    }
```

Reply:
though I am worried that this flow is starting to get messy like the former flow we deprecated. Perhaps it would be beneficial to break this function up into smaller functions?

Reply:
I think breaking this function up just for the uiPredicates & entitlements might be too much, I merged the two value into one and simplified the code a little, let me know if this makes sense, thanks!